src/modules/html.c

```diff
@@ -60,20 +60,20 @@ static unsigned int html_reply(struct http_request *request, void *data)
 	GET_PRIV(data, html);
 	const char *url_stub = (strlen(request->url) > strlen("/html/")) ? request->url + strlen("/html/") : "index.html";
 	if (url_stub[0] == '/' || strstr(url_stub,"/../") || !strncmp(url_stub,"../",strlen("../"))) {
-		send_response_fail(request->connection, "Invalid URL");
+		send_response_bad_request(request->connection, "Invalid URL");
 		return 0;
 	}
 	snprintf(path, sizeof(path), "%s/%s", core->config->H_arg, url_stub);
 	ret = stat(path, &sbuf);
 	if (ret < 0) {
 		warnlog(html->logger, "Stat failed for %s. Errnno %d: %s.", path,errno,strerror(errno));
-		send_response_fail(request->connection, "stat() was not happy");
```

Comment:
This is wrong. stat ()  can fail for any number of reasons. Like permission issues in the path.